aten/src/ATen/native/Resize.cpp

```diff
@@ -306,6 +306,28 @@ void resize_bytes_nocuda(const Storage& storage, const c10::SymInt& newsize) {
     // behavior when compared to resize_bytes_xxx. For these cases,
     // an additional memory copy and update for storage are required.
     if (original_data_ptr == src_tensor.storage().data_ptr().get()) {
+      // FIXME: ideally we should call resize_bytes_mtia() directly in this function
```

Comment:
This kind of device specific logic is NOT ok.
Please appropriately refactor the code or register a different op in native_functions.yaml if you need something different from the generic impl.